src/test/java/com/aws/greengrass/logmanager/model/LogFileTest.java

```diff
@@ -25,53 +28,86 @@ public class LogFileTest {
-                fileOutputStream.write(String.valueOf(i + 1).getBytes(StandardCharsets.UTF_8));
-                fileOutputStream.write("\n".getBytes(StandardCharsets.UTF_8));
-            }
+            fileOutputStream.write(byteArray);
         }
     }
 
     @Test
     void GIVEN_empty_file_WHEN_calculate_file_hash_THEN_we_get_null() throws IOException {
         File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
+        byte[] bytesArray = getRandomTestStrings(0).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
         LogFile logFile = LogFile.of(file);
         String fileHash = logFile.hashString();
         assertEquals(fileHash, HASH_VALUE_OF_EMPTY_STRING);
+        file.delete();
     }
 
+    @Test
+    void GIVEN_log_file_with_less_than_target_lines_WHEN_calculate_file_hash_THEN_we_get_null()
+            throws IOException {
+        File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM - 100).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
+        LogFile logFile = LogFile.of(file);
+        String fileHash = logFile.hashString();
+        assertEquals(fileHash, HASH_VALUE_OF_EMPTY_STRING);
+        file.delete();
+    }
+
+
     @Test
     void GIVEN_log_file_with_equal_to_target_lines_WHEN_calculate_file_hash_THEN_we_get_null()
             throws IOException, NoSuchAlgorithmException {
         File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
-        writeFiles(file, DEFAULT_LINES_FOR_DIGEST_NUM);
-        StringBuilder msg = new StringBuilder();
-        for (int i = 0; i < DEFAULT_LINES_FOR_DIGEST_NUM; i++) {
-            msg.append("line").append(i + 1);
-        }
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
         LogFile logFile = LogFile.of(file);
         String fileHash = logFile.hashString();
-        assertEquals(fileHash, calculate(msg.toString()));
+        String msg = new String(bytesArray);
+        assertEquals(fileHash, calculate(msg));
         file.delete();
     }
 
     @Test
     void GIVEN_log_file_with_more_than_target_lines_WHEN_calculate_file_hash_THEN_we_get_hash()
             throws IOException, NoSuchAlgorithmException {
         File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
-        writeFiles(file, DEFAULT_LINES_FOR_DIGEST_NUM + 1);
-        StringBuilder msg = new StringBuilder();
-        for (int i = 0; i < DEFAULT_LINES_FOR_DIGEST_NUM; i++) {
-            msg.append("line").append(i + 1);
-        }
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM + 100).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
+        LogFile logFile = LogFile.of(file);
```

Comment:
Why do this? You could create a LogFile directly since it is just a file. 

```
LogFile file = new LogFile(directoryPath.resolve("greengrass_test.log").toUri());
```